pkg/controller/installation/core_controller.go

```diff
@@ -1429,6 +1431,28 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 		FelixPrometheusMetricsEnabled: utils.IsFelixPrometheusMetricsEnabled(felixConfiguration),
 		FelixPrometheusMetricsPort:    felixPrometheusMetricsPort,
 	}
+
+	// Check if BPF auto-bootstrap is enabled and its requirements.
+	bpfBootstrapReq, err := bpfAutoBootstrapRequirements(r.client, ctx, instance, felixConfiguration)
+	if err != nil {
+		r.status.SetDegraded(operator.ResourceValidationError, "Unable to retrieve BPF auto-bootstrap requirements", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// If BPF auto-bootstrap requirements are met, configure the node and disable kube-proxy.
+	if bpfBootstrapReq != nil {
+		// Extract k8s service and endpoints to push them to ebpf-bootstrap init container.
+		nodeCfg.K8sServiceAddrs = serviceIPsAndPorts(bpfBootstrapReq.k8sService)
+		nodeCfg.K8sEndpointSlice = serviceEndpointSlice(bpfBootstrapReq.k8sServiceEndpoints)
+
+		// Try to disable kube-proxy.
+		err = r.disableKubeProxy(ctx, felixConfiguration, bpfBootstrapReq.kubeProxyDs)
```

Comment:
I asked Casey in the design and he suggested you should add a new controller for updating the kubeproxy daemonset.